DataFormats/src/PATFinalState.cc

```diff
@@ -780,6 +799,33 @@ PATFinalState::subcand(int i, int j, int x, int y, int z) const {
       new PATMultiCandFinalState(output, evt()));
 }
 
+
+PATFinalStateProxy
+PATFinalState::subcand(int i, const std::string&  tagi,int j, const std::string&  tagj, int x, int y, int z) const {
+  std::vector<reco::CandidatePtr> output;
+ 
+ if(daughterHasUserCand(i,tagi)){
+ output.push_back(daughterUserCand(i,tagi));
+ } else { output.push_back(daughterPtr(i));}
+ if(daughterHasUserCand(j,tagj)){
+ output.push_back(daughterUserCand(j,tagj));
+ } else { output.push_back(daughterPtr(j));}
+
+  if (x > -1)
+    output.push_back(daughterPtr(x));  // not touching this, I think we only want this for the pt/eta/mass, not the internal 
```

Comment:
I just realized the comment is wrong. I can make this more general including more tags for x,y,z